server/service/src/purchase_order/generate.rs

```diff
@@ -0,0 +1,70 @@
+use repository::{
+    EqualFilter, ItemRowRepository, NumberRowType, PurchaseOrderLineRow, PurchaseOrderRow,
+    RepositoryError, StockOnHandFilter, StockOnHandRepository,
+};
+use util::uuid::uuid;
+
+use crate::number::next_number;
+use crate::service_provider::ServiceContext;
+use crate::store_preference::get_store_preferences;
+
+pub fn generate_empty_purchase_order_lines(
+    ctx: &ServiceContext,
+    purchase_order_row: &PurchaseOrderRow,
+    item_ids: Vec<String>,
+) -> Result<Vec<PurchaseOrderLineRow>, RepositoryError> {
+    let mut result: Vec<PurchaseOrderLineRow> = Vec::new();
+    let store_preferences = get_store_preferences(&ctx.connection, &purchase_order_row.store_id)?;
+
+    let mut line_number = next_number(
+        &ctx.connection,
+        &NumberRowType::PurchaseOrderLine(purchase_order_row.id.clone()),
+        &purchase_order_row.store_id,
+    )?;
+
+    let stocks_on_hand = StockOnHandRepository::new(&ctx.connection).query(Some(
```

Comment:
I would probably put this into HashMap rather than array, although it doesn't really matter.